lib/getCommitAuthor.js

```diff
@@ -0,0 +1,30 @@
+const token = process.env.GITHUB_TOKEN;
+const Octokit = require('@octokit/rest');
+
+const octokit = new Octokit({
+  auth: `token ${token}`,
+});
+
+async function getCommitAuthor(build) {
+  const cloudSourceRepo = build.source.repoSource.repoName;
+  const { commitSha } = build.sourceProvenance.resolvedRepoSource;
+
+  // format github_ownerName_repoName
+  const [, githubOwner, githubRepo] = cloudSourceRepo.split('_');
+
+  try {
+    // get github commit object
+    const githubCommit = await octokit.git.getCommit({
+      commit_sha: commitSha,
+      owner: githubOwner,
+      repo: githubRepo,
+    });
+    // return github commit author e-mail
+    const commitAuthor = githubCommit.data.author.email;
+    return commitAuthor;
+  } catch (error) {
+    console.log(error);
```

Comment:
The error should be reported to the caller

Reply:
refactored code, please review